### PR TITLE
Nicer error handling for selected modules

### PR DESCRIPTION
**Pull request by ronpal**
https://cognitedata.atlassian.net/browse/CDF-21260

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py [hunk 1]
```diff
@@ -48,7 +48,9 @@ def load(cls, data: dict[str, Any], build_name: str) -> Environment:
                     if MODULE_PATH_SEP in selected
                     else selected
                     for selected in data["selected_modules_and_packages"]
-                ],
+                ]
+                if data["selected_modules_and_packages"]
+                else [],
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: Not sure what this achieves, it looks like the same as was already there. Do you try to make it optional to provide |selected_modules_and_packages`?

Comment (doctrino) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: What about just changing the line above to `data.get("selected_modules_and_packages", [])`?

**Review by doctrino — COMMENTED** [review]
Should also have a little entry in the changelog

File: cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py (hunk 1 above)
Comment (ronpal) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: This code covers the rare case when you have the key "selected_modules_and_packages" but no items beneath it, I don't think yours does?

Comment (ronpal) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pal.ronning@cognitedata.com/Local/cdf-project-templates/cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py", line 133, in load
    environment = Environment.load(data["environment"], build_env_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pal.ronning@cognitedata.com/Local/cdf-project-templates/cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py", line 50, in load
    for selected in data["selected_modules_and_packages", []]
                    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'

Comment (doctrino) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: Ah now I get it. I assumed `pyyaml` parsed an empty list, but of course it cannot now whether it is a list or a mapping, so it parses None.

I though you had this
```python
data = {"selected_modules_and_packages": []}
```
while you actually got this:
```python
data = {"selected_modules_and_packages": None}
```

Comment (doctrino) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: Anyways if you want to 🏌

then , this also works: 
```python
for selected in data["selected_modules_and_packages"] or []
```

**Review by doctrino — APPROVED** [review]
Pre-approving, but please include an entry in the changelog. I thin `### Improvements` is the appropriate heading.

File: cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py (hunk 1 above)
Comment (ronpal) [reply cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py:53]: 🏌️ ⛳

**Pull request merged**